jbehave-core/src/main/java/org/jbehave/core/steps/ParameterConverters.java

```diff
@@ -320,6 +322,27 @@ public Object convert(String value, Type type, Story story) {
             }
         }
 
+        if (type instanceof Class) {
+            Class clazz = (Class) type;
+            if (clazz.isArray()) {
+                String[] values = value.isEmpty() ? new String[0] : value.split(escapedCollectionSeparator);
+                Class componentType = clazz.getComponentType();
+                ParameterConverter elementConverter = findConverter(componentType);
+                Object array = null;
+                
+                try {
+                    array = Array.newInstance(componentType, values.length);
+                } catch (Throwable e) {
+                    // Could not instantiate array, swallowing exception quietly
+                }
```

Comment:
what about moving lines 331-337 to a new method `createArray`? (it will make the entire method not so complex, and again the same logical pattern as for collections)